Done.
Setting up the random seed for reproducibility...
Done.
dynamic_import - class_name:  OracleTrainer
Initializing model...
Traceback (most recent call last):
  File "/home/utente/pycharm/plugins/python-ce/helpers/pydev/pydevd.py", line 1496, in _exec
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "/home/utente/pycharm/plugins/python-ce/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "/home/utente/Scrivania/PROVA/LADD/src/run.py", line 53, in <module>
    run_experiment()
  File "/home/utente/Scrivania/PROVA/LADD/src/run.py", line 15, in run_experiment
    trainer = trainer_class(args, writer, device, rank, world_size)
  File "/home/utente/Scrivania/PROVA/LADD/src/federated/trainers/oracle_trainer.py", line 8, in __init__
    super().__init__(args, writer, device, rank, world_size)
  File "/home/utente/Scrivania/PROVA/LADD/src/federated/trainers/trainer.py", line 11, in __init__
    super().__init__(args, writer, device, rank, world_size)
  File "/home/utente/Scrivania/PROVA/LADD/src/general_trainer.py", line 44, in __init__
    self.clients_args = DatasetHandler(args, writer)()
  File "/home/utente/Scrivania/PROVA/LADD/src/utils/data_utils.py", line 24, in __init__
    self.__target_dataset_init()
  File "/home/utente/Scrivania/PROVA/LADD/src/utils/data_utils.py", line 165, in __target_dataset_init
    ds = self.__gen_ds(split_data[user], dataset_name, dataset, train_transform, test_transform,
  File "/home/utente/Scrivania/PROVA/LADD/src/utils/data_utils.py", line 119, in __gen_ds
    for rgb_path, hha_path in zip(rgb_paths, hha_paths):
UnboundLocalError: local variable 'rgb_paths' referenced before assignment
Done.
Initializing datasets...